diff --git a/ffbuild/library.mak b/ffbuild/library.mak
index ad09f20..e63196d 100644
--- a/ffbuild/library.mak
+++ b/ffbuild/library.mak
@@ -35,7 +35,9 @@ OBJS += $(SHLIBOBJS)
 endif
 $(SUBDIR)$(LIBNAME): $(OBJS) $(STLIBOBJS)
 	$(RM) $@
-	$(AR) $(ARFLAGS) $(AR_O) $^
+	$(file >$@.ar.txt, $^)
+	$(AR) $(ARFLAGS) $(AR_O) @$@.ar.txt
+	$(RM) $@.ar.txt
 	$(RANLIB) $@
 
 install-headers: install-lib$(NAME)-headers install-lib$(NAME)-pkgconfig
